src/Extractor/Rst.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rusty\Extractor;
+
+use Gregwar\RST\ErrorManager;
+use Gregwar\RST\HTML\Nodes\CodeNode;
+use Gregwar\RST\Parser;
+use League\CommonMark\Block\Element;
+use Rusty\CodeSample;
+use Rusty\PragmaParser;
+
+class Rst implements SampleExtractor
+{
+    protected $pragmaParser;
+
+    public function __construct()
+    {
+        $this->pragmaParser = new PragmaParser();
+    }
+
+    public static function supportedExtensions(): array
+    {
+        return ['rst'];
+    }
+
+    public function extractSamples(\SplFileInfo $file): \Traversable
+    {
+        $parser = new Parser();
+
+        $parser->getEnvironment()->setErrorManager(new class() extends ErrorManager {
```

Comment:
Because some directives are not supported by the parser (we only need the code example directives). And when it appends an ugly echo is made : https://github.com/Gregwar/RST/blob/master/ErrorManager.php#L19. This anonymous class prevents that.